crates/torii/src/processors/store_set_record_state_diff.rs

```diff
@@ -0,0 +1,59 @@
+use crate::state::State;
+use anyhow::{Error, Ok, Result};
+use async_trait::async_trait;
+use chrono::offset;
+use dojo_types::component;
+use starknet::core::types::BlockId;
+use starknet::core::types::StateDiff;
+use starknet::core::utils::cairo_short_string_to_felt;
+use starknet::core::utils::parse_cairo_short_string;
+use starknet::macros::short_string;
+use starknet_crypto::poseidon_hash_many;
+use starknet_crypto::FieldElement;
+use tracing::info;
+
+use super::StateDiffProcessor;
+
+#[derive(Default)]
+pub struct StoreSetRecordStateDiffProcessor;
+
+#[async_trait]
+impl<S: State + std::marker::Sync> StateDiffProcessor<S> for StoreSetRecordStateDiffProcessor {
```

Comment:
We can make this an implementation of `BlockProcessor`.

It takes the block as input, retrieves the state diff for the world and sets the values. We'll need to update the `set_entity` api to operate on the storage key. I think an approach that might make sense it to modifiy `StoreSetRecord` to only call `create_entity` which will store the storage key preimage and then set the values directly using the state diff.

Does that make sense?